sdk-core/src/main/java/care/data4life/sdk/ApiService.java

```diff
@@ -245,12 +293,37 @@ Completable deleteRecord(String alias, String recordId, String userId) {
                 .subscribeOn(Schedulers.io());
     }
 
+    /**
+     * Carry out needed logout actions.
+     * <p>
+     * When using refresh token, this will revoke the OAuth access.
+     * When using a static access token, nothing will be done.
+     *
+     * @param alias Alias
+     * @return Completable
+     */
     Completable logout(String alias) {
+        if (this.useStaticToken) {
+            throw new D4LRuntimeException("Cannot log out when using a static access token!");
+        }
         return Single
                 .fromCallable(() -> oAuthService.getRefreshToken(alias))
                 .flatMapCompletable(token -> service.logout(alias, token));
+
     }
 
+    /**
```

Comment:
it's quite unusual to have a documentation on private class members

Reply:
Right, this would be more for internal documentation purposes, I wrote it up to capture my understanding of the function. Would you prefer to have it removed? (e.g. for consistency or if it messes up automatic generation of documentation by including private classes or for consistency)